oxidation/libparsec/crates/types/src/regex.rs

```diff
@@ -12,43 +12,110 @@ fn escape_globing_pattern(string: &str) -> String {
 }
 
 impl Regex {
+    pub fn from_patterns_file(file_content: &str) -> Result<Self, fnmatch_regex::error::Error> {
```

Comment:
```suggestion
    pub fn from_patterns(raw_regexes: &[str]) -> Result<Self, fnmatch_regex::error::Error> {
```
I think it would be better to pass a list of raw regexes :thinking:

Reply:
Yes, it means that we should call `splitlines` in python. Isn't it better to just take the file path as parameter and handle everything in rust?

Reply:
You can create both `from_file(path: &Path)` and `from_raw_regexes(raw_regexes: &[str])`

Reply:
I've implemented both methods and improved error handling